WebContent/util/serverStatusCheck.html

```diff
@@ -0,0 +1,120 @@
+<HTML>
+<HEAD><TITLE>Website Service Status</TITLE></HEAD>
+
+<SCRIPT>
+var xmlHttp = false;
+
+	//http://www.ibm.com/developerworks/web/library/wa-ajaxintro1/index.html
+	/* Create a new XMLHttpRequest object to talk to the Web server */
+	xmlHttp = false;
+	/*@cc_on @*/
+	/*@if (@_jscript_version >= 5)
+	try {
+	  xmlHttp = new ActiveXObject("Msxml2.XMLHTTP");
+	} catch (e) {
+	  try {
+	    xmlHttp = new ActiveXObject("Microsoft.XMLHTTP");
+	  } catch (e2) {
+ 	   xmlHttp = false;
+ 	 }
+	}
+	@end @*/
+
+	
+	if (!xmlHttp && typeof XMLHttpRequest != 'undefined') {
+	  xmlHttp = new XMLHttpRequest();
+	}
+
+var sLastHostName='';
+function checkSiteStatus(sHostName)
+{
+	sLastHostName = sHostName;
+	//Make JSON request
+	xmlHttp.open("GET","util/serverStatusCheckService.jsp?HostName=" + sHostName);
+	xmlHttp.onreadystatechange = StateChangeForJSON;
+	xmlHttp.send(null);
+}
+function StateChangeForJSON()
+{
+	if(xmlHttp.readyState == 4 && xmlHttp.status == 200)
+	{
+		var jsonObj = eval('('+ xmlHttp.responseText + ')');
+		var jsonFetchHostStatus = jsonObj["HostStatus"];
+		var jsonFetchHostName=jsonObj["HostName"];
+		//get JSON values and output
+		x=document.getElementById('FetchHostName');
+		x.innerHTML=jsonFetchHostName;
+		x=document.getElementById('FetchHostStatus');
+		x.innerHTML=jsonFetchHostStatus;
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg" width="20"/></sub>&nbsp; Cross-site Scripting (XSS)

Unsanitized input from `data from a remote resource` flows into `innerHTML`, where it is used to dynamically construct the HTML page on client side. This may result in a DOM Based Cross-Site Scripting attack (DOMXSS).

<h6>
Line 48 | <a href="https://cwe.mitre.org/data/definitions/79.html">CWE-79</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 553</a> | <a href="https://learn.snyk.io/lesson/xss/">Learn more about this vulnerability</h6>

<details>

<summary><b>Data flow: 8 steps</b></summary>

`Step 1 - 4` https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/util/serverStatusCheck.html#L41

`Step 5 - 6` [WebContent/util/serverStatusCheck.html#L42](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/util/serverStatusCheck.html#L42)

`Step 7 - 8` https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/util/serverStatusCheck.html#L48

</details>